jps-plugin/src/org/intellij/erlang/jps/builder/ErlangBuilderService.java

```diff
@@ -16,26 +16,26 @@
 
 package org.intellij.erlang.jps.builder;
 
+import com.intellij.util.containers.ContainerUtil;
+import org.intellij.erlang.jps.rebar.RebarBuilder;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.jps.builders.BuildTargetType;
 import org.jetbrains.jps.incremental.BuilderService;
 import org.jetbrains.jps.incremental.TargetBuilder;
-import org.intellij.erlang.jps.rebar.RebarBuilder;
 
 import java.util.Arrays;
-import java.util.Collections;
 import java.util.List;
 
 public class ErlangBuilderService extends BuilderService {
   @NotNull
   @Override
   public List<? extends BuildTargetType<?>> getTargetTypes() {
-    return Collections.singletonList(ErlangTargetType.INSTANCE);
+    return ContainerUtil.newArrayList(ErlangPrepareDependenciesTargetType.INSTANCE, ErlangTargetType.INSTANCE);
   }
 
   @NotNull
   @Override
   public List<? extends TargetBuilder<?, ?>> createBuilders() {
-    return Arrays.asList(new ErlangBuilder(), new RebarBuilder());
+    return Arrays.asList(new ErlangPrepareDependenciesBuilder(), new ErlangBuilder(), new RebarBuilder());
```

Comment:
So you've changed Collections usages to ContainerUtil usages, yet left Arrays usages untouched. Why?